src/hbbft.erl

```diff
@@ -461,3 +490,22 @@ group_by([], D) ->
     lists:keysort(1, [{K, lists:sort(V)} || {K, V} <- dict:to_list(D)]);
 group_by([{K, V}|T], D) ->
     group_by(T, dict:append(K, V, D)).
+
+
+proposed(Data = #hbbft_data{n=N, batch_size=BatchSize, buf=Buf}) ->
+    Proposed = hbbft_utils:random_n(min((BatchSize div N), length(Buf)), lists:sublist(Buf, BatchSize)),
```

Comment:
I have mixed feelings about this.  Is there a way for the transaction to become invalid while sitting in the buffer?  If not, can't we just filter on the way into the buffer?  If there is, this seems kind of unpredictable, depending on the cost of the validation.  Depending on how big we set `BatchSize`, it might be worth the complexity to accumulate (and shrink the subsequent N for pulling from the buffer), rather than starting over each time.

Reply:
This is a very fair point, maybe it is better to simply do that and allow transactions that become invalid at rest to simply be purged by consensus once the block is proposed.